File: app/src/main/java/com/nimtego/plectrum/data/repository/datasource/DataStore.java
package com.nimtego.plectrum.data.repository.datasource;

import com.nimtego.plectrum.data.model.rss_itunes.Feed;
import com.nimtego.plectrum.data.rest.pojo.AlbumsRepository;
import com.nimtego.plectrum.data.rest.pojo.ArtistsRepository;
import com.nimtego.plectrum.data.rest.pojo.SongsRepository;
import com.nimtego.plectrum.data.rest.pojo.wiki.WikiSearchResult;

import io.reactivex.Observable;

public interface DataStore {

    Observable<WikiSearchResult> wikiSearch(String response);

    Observable<SongsRepository> songs(String response);

    Observable<SongsRepository> songsByIdAlbum(int id);

    Observable<ArtistsRepository> artists(String response);

    Observable<AlbumsRepository> albums(String response);

    Observable<SongsRepository> songById(int id);

    Observable<ArtistsRepository> artistById(int id);

    Observable<AlbumsRepository> album(String response);

    Observable<Feed> recent();

    Observable<Feed> topSong();

    Observable<Feed> topAlbum();

    Observable<Feed> hot();

    Observable<Feed> newMusick();
}

File: app/src/main/java/com/nimtego/plectrum/data/repository/datasource/DataStore.kt
package com.nimtego.plectrum.data.repository.datasource

import com.nimtego.plectrum.data.model.rss_itunes.PopularResponse
import com.nimtego.plectrum.data.model.itunes.AlbumsRepository
import com.nimtego.plectrum.data.model.itunes.ArtistsRepository
import com.nimtego.plectrum.data.model.itunes.SongsRepository
import com.nimtego.plectrum.data.model.wiki.WikiSearchResult

import io.reactivex.Observable

@Deprecated("<ISP. Remove when stops being useful>")
interface DataStore {

    fun wikiSearch(response: String): Observable<WikiSearchResult>

    fun songs(response: String): Observable<SongsRepository>

    fun songsByIdAlbum(id: Int): Observable<SongsRepository>

    fun artists(response: String): Observable<ArtistsRepository>

    fun albums(response: String): Observable<AlbumsRepository>

    fun songById(id: Int): Observable<SongsRepository>

    fun artistById(id: Int): Observable<ArtistsRepository>

    fun album(response: String): Observable<AlbumsRepository>

    fun recent(): Observable<PopularResponse>

    fun topSong(size: Int): Observable<PopularResponse>

    fun topAlbum(): Observable<PopularResponse>

    fun hot(): Observable<PopularResponse>

    fun newMusic(): Observable<PopularResponse>

    fun topMovie(): Observable<PopularResponse>

    fun topFreeBooks(): Observable<PopularResponse>

    fun topPaidBooks(): Observable<PopularResponse>
}
